object_manager.go

```diff
@@ -562,6 +562,66 @@ func (objMgr *ObjectManager) DeleteCNAMERecord(ref string) (string, error) {
 	return objMgr.connector.DeleteObject(ref)
 }
 
+func (objMgr *ObjectManager) CreateTXTRecord(recordname string, text string, dnsview string) (*RecordTXT, error) {
+
+	recordTXT := NewRecordTXT(RecordTXT{
+		View: dnsview,
+		Name: recordname,
+		Text: text,
+	})
+
+	ref, err := objMgr.connector.CreateObject(recordTXT)
+	recordTXT.Ref = ref
+	return recordTXT, err
+}
+
+func (objMgr *ObjectManager) GetTXTRecordByRef(ref string) (*RecordTXT, error) {
+	recordTXT := NewRecordTXT(RecordTXT{})
+	err := objMgr.connector.GetObject(recordTXT, ref, &recordTXT)
+	return recordTXT, err
+}
+
+func (objMgr *ObjectManager) GetTXTRecord(name string) (*RecordTXT, error) {
+	if len(name) == 0 {
+		return nil, nil
+	}
+	var res []RecordTXT
+
+	recordTXT := NewRecordTXT(RecordTXT{Name: name})
+
+	err := objMgr.connector.GetObject(recordTXT, "", &res)
+
+	if err != nil || res == nil || len(res) == 0 {
+		return nil, err
+	}
+
+	return &res[0], nil
+}
+
+func (objMgr *ObjectManager) UpdateTXTRecord(recordname string, text string) (*RecordTXT, error) {
+	var res []RecordTXT
+
+	recordTXT := NewRecordTXT(RecordTXT{Name: recordname})
+
+	err := objMgr.connector.GetObject(recordTXT, "", &res)
+
+	res[0].Text = text
+
+	res[0].Zone = "" // otherwize infoblox API freaks out with ""text": "Field is not writable: zone""
```

Comment:
by default it must be "" if you dont mention it, wondering why it needs to be explicitly mentioned ?

Reply:
@chinmayb exactly,  if Zone is set or inherited from retrieved object then update will fail with the message from the comment

Reply:
got it, please update the description 
```suggestion
	res[0].Zone = "" //  set the Zone value to "" as its a non writable field
```

Reply:
@chinmayb totally, thanks, done